com.unity.perception/Tests/Performance/PerformanceTester.cs

```diff
@@ -0,0 +1,139 @@
+        {
+            switch (label) {
+                case PerformanceTestObjectCountLabeler.Label:
+                    return new ObjectCountLabeler(idConfig);
+                case PerformanceTestBoundingBoxLabeler.Label:
+                    return new BoundingBox2DLabeler(idConfig);
+                case PerformanceTestPixelCountLabeler.Label:
+                     return new RenderedObjectInfoLabeler(idConfig);
+                case PerformanceTestSemanticSegmentationLabeler.Label:
+                     return new SemanticSegmentationLabeler(ssConfig);
+
+                default:
+                    return null;
+            }
+        }
+
+        private static List<CameraLabeler> CreateLabelers(IEnumerable<string> labelers, IdLabelConfig idConfig, SemanticSegmentationLabelConfig ssConfig)
+        {
+            return labelers.Select(l => CreateLabeler(l, idConfig, ssConfig)).Where(labeler => labeler != null).ToList();
+        }
+
+        [SetUp]
+        public void SetUpTest()
+        {
+            DatasetCapture.ResetSimulation();
+            Time.timeScale = 1;
+
+            if (Directory.Exists(DatasetCapture.OutputDirectory))
+                Directory.Delete(DatasetCapture.OutputDirectory, true);
+
+            Screen.SetResolution(m_Resolution.Item1, m_Resolution.Item2, true);
+            (m_Camera, m_IdConfig, m_SsConfig, m_SceneRoot) = TestHelper.CreateThreeBlockScene();
+
+            m_Camera.enabled = true;
+            m_Camera.showVisualizations = false;
+
+            m_ActiveLabelers = CreateLabelers(m_Labelers, m_IdConfig, m_SsConfig);
+
+            foreach (var l in m_ActiveLabelers)
+            {
+                m_Camera.AddLabeler(l);
+            }
+
+            if (!m_CaptureData) m_Camera.enabled = false;
+            if (m_CaptureData && !m_VisualizersOn) m_Camera.showVisualizations = false;
+            m_Camera.gameObject.SetActive(true);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            Object.DestroyImmediate(m_Camera.gameObject);
+            Object.DestroyImmediate(m_SceneRoot.gameObject);
+
+            var simState = DatasetCapture.SimulationState;
+            simState.End();
+
+            DatasetCapture.ResetSimulation();
+            Time.timeScale = 1;
+            if (Directory.Exists(DatasetCapture.OutputDirectory))
+                Directory.Delete(DatasetCapture.OutputDirectory, true);
+
+            m_ActiveLabelers = null;
+            m_IdConfig = null;
+            m_SsConfig = null;
+        }
+
+        [UnityTest, Performance]
+        public IEnumerator ExecuteTest()
+        {
+            yield return Measure.Frames()
+                .WarmupCount(10)
+                .MeasurementCount(30)
+                .Run();
+
+            // Allow all file writes to complete
+            yield return new WaitForSeconds(5);
```

Comment:
DatasetCapture.ResetSimulaton() will take care of this

Reply:
If it doesn't then we should look into it later :)

Reply:
Ok, I'll try that out. I actually didn't know about that call until recently, I had to put it in to fix some test errors.

Reply:
It may be something that we look into again later. When I removed it, the standard deviation and max execution time rose quite a bit when running with a series of tests executing one after another.